src/output-json-mqtt.c

```diff
@@ -89,8 +81,10 @@ static int JsonMQTTLogger(ThreadVars *tv, void *thread_data,
         return TM_ECODE_FAILED;
     }
 
-    if (!rs_mqtt_logger_log(state, tx, thread->mqttlog_ctx->flags, js))
+    jb_open_object(js, "mqtt");
+    if (!rs_mqtt_logger_log(tx, thread->mqttlog_ctx->flags, js))
         goto error;
```

Comment:
no close on error?

Reply:
No, the error handler just free's it.  Its not used anymore.

Reply:
Indeed as Jason said.

To be more complete :
- When we log a MQTT event, if we fail to log the MQTT data, we do not log anything at all
- When we log an alert, if we fail to log the MQTT data, we still log the alert